src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -27,6 +28,132 @@ Onyx.connect({
+// apply the updates in order after the missing updates are fetched and applied
+function validateAndApplyDeferredUpdates(): Promise<void> {
+    // We only want to apply deferred updates that are newer than the last update that was applied to the client.
+    // At this point, the missing updates from "GetMissingOnyxUpdates" have been applied already, so we can safely filter out.
+    const pendingDeferredUpdates = Object.fromEntries(
+        Object.entries(deferredUpdates).filter(([lastUpdateID]) => {
+            // It should not be possible for lastUpdateIDAppliedToClient to be null,
+            // after the missing updates have been applied.
+            // If still so we want to keep the deferred update in the list.
+            if (!lastUpdateIDAppliedToClient) {
+                return true;
+            }
+            return (Number(lastUpdateID) ?? 0) > lastUpdateIDAppliedToClient;
+        }),
+    );
+    const pendingDeferredUpdateValues = Object.values(pendingDeferredUpdates);
+
+    // If there are no remaining deferred updates after filtering out outdated ones,
+    //  we can just unpause the queue and return
+    if (pendingDeferredUpdateValues.length === 0) {
+        return Promise.resolve();
+    }
+
+    const {applicableUpdates, updatesAfterGaps, latestMissingUpdateID} = detectGapsAndSplit(pendingDeferredUpdates);
+
+    // If we detected a gap in the deferred updates, only apply the deferred updates before the gap,
+    // re-fetch the missing updates and then apply the remaining deferred updates after the gap
+    if (latestMissingUpdateID) {
+        return new Promise((resolve, reject) => {
+            deferredUpdates = {};
+            applyUpdates(applicableUpdates).then(() => {
+                // Same as above, we can ignore this case because
+                // it should not be possible for lastUpdateIDAppliedToClient to be null.
+                if (!lastUpdateIDAppliedToClient) {
+                    return;
+                }
+
+                // After we have applied the applicable updates, there might have been new deferred updates added.
+                // In the next (recursive) call of "validateAndApplyDeferredUpdates",
+                // the initial "updatesAfterGaps" and all new deferred updates will be applied in order,
+                // as long as there was no new gap detected. Otherwise repeat the process.
+                deferredUpdates = {...deferredUpdates, ...updatesAfterGaps};
+
+                // Then we can fetch the missing updates and apply them
+                App.getMissingOnyxUpdates(lastUpdateIDAppliedToClient, latestMissingUpdateID)
+                    .then(validateAndApplyDeferredUpdates)
+                    .then(() => resolve())
+                    .catch(reject);
+            });
+        });
+    }
+
+    // If there are no gaps in the deferred updates, we can apply all deferred updates in order
+    return applyUpdates(applicableUpdates).then(() => undefined);
```

Comment:
why are we doing this `.then(() => undefined)`?

Reply:
Anyway, i changed the type of `queryPromise` to reflect the actual return values